hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/MembershipUpdateTest.java

```diff
@@ -735,6 +738,30 @@ public void noOperationExecuted_beforePreJoinOpIsDone() throws Exception {
         assertTrue(service.preJoinOpExecutionCompleted.get());
     }
 
+    @Test
+    public void testListenerRegistrationWhileANewMemberJoining() {
+        CountDownLatch latch = new CountDownLatch(1);
+
+        Config masterConfig = getConfigWithService(new PostJoinAwareServiceImpl(latch), PostJoinAwareServiceImpl.SERVICE_NAME);
+        HazelcastInstance hz1 = factory.newHazelcastInstance(masterConfig);
+
+        Config joiningConfig = getConfigWithService(new PostJoinAwareServiceImpl(latch), PostJoinAwareServiceImpl.SERVICE_NAME);
+        EntryListenerConfig listenerConfig = new EntryListenerConfig();
+        CountDownLatch entryAddedLatch = new CountDownLatch(1);
+        listenerConfig.setImplementation((EntryAddedListener) event -> entryAddedLatch.countDown());
+        joiningConfig.getMapConfig("test").addEntryListenerConfig(listenerConfig);
+
+        factory.newHazelcastInstance(joiningConfig);
```

Comment:
This test seems to have a wrong base because we always assume same config for all members when starting a cluster, otherwise it is possible to create such many failure cases.

Reply:
Updated the test, thanks for the note!